src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -162,14 +172,14 @@ offset_index::find_kaf_offset(model::offset upper_bound) {
       _file_index.get_row_count(),
       _file_index.copy());
     auto file_pos = _fetch_ix(std::move(file_dec), ix);
-    res.file_pos = *file_pos;
+    res.file_pos = *file_pos + _file_pos_sampling_step;
     return res;
 }
 
 struct offset_index_header
   : serde::envelope<
       offset_index_header,
-      serde::version<0>,
+      serde::version<1>,
```

Comment:
do we need to update compat version also?

Reply:
I don't think so, old fields are still present, so old version will be able to read it
also, it's not part of any release so we will never see indexes created by old version

Reply:
if it isn't part of a release, is there any reason to bump the version yet?

it's in our 22.1.1 beta releases, fwiw

Reply:
I think that @LenaAn has a good question (although it is true that we haven't shipped this so it's probably not going to affect anything).

> I don't think so, old fields are still present, so old version will be able to read it

I think this is true, but is it also a bigger question than if an old version could decode the header, and relate to what an old version of the code would do (that doesn't understand sampling) if it were to read data built by code that implemented sampling.